code/headers/display_adapter.hpp

```diff
@@ -61,13 +87,125 @@ namespace r2d2::display {
+                    set_pixel(x + t_x, y + t_y, data);
+                }
+            }
+        }
+
+
+        /**
+         * @brief Sets character in a single color
+         * 
+         * @param x x-coordinate of the character (x=0 is the leftmost collumn)
+         * 
+         * @param y y-coordinate of the character (y=0 is the highest row)
+         * 
+         * @param character The un-extended (0-127) ascii value of the character
+         *  
+         * 
+         * @param pixel_color The color of the pixel
+         * 
+         */
+        virtual void set_character(uint16_t x, uint16_t y, char character, 
+                                uint16_t pixel_color) {
+            // Loops through all rows of the character 
+            const hwlib::image& character_image = display_font[character];                               
+            for (int image_y = 0; image_y < 8; image_y++) {
+                for (int image_x = 0; image_x < 8; image_x++) {
+                    // Check if the pixel needs to be drawn. 
+                    // If the pixel color is anything other than white, draw it
+                    hwlib::color character_pixel_color = character_image[hwlib::xy(image_x, image_y)];
+                    if (character_pixel_color != hwlib::white) {
+                        set_pixel(x + image_x, y + image_y, pixel_color);
+                    }
+                }
+            }
+        }
+
+        /**
+         * @brief Draws given characters to the target cursor. For every 
+         * character drawn this way, the cursor will move 8 pixels.
+         * 
+         * @param cursor_target This targets the cursor with which to draw
+         * 
+         * @param characters Array of characters to draw
+         * 
+         * @param character_amount number of characters to draw
+         */
+        virtual void set_character(uint8_t cursor_target, const char *characters){
+            // Checks if the given cursor is not out of bounds
+            if (cursor_target >= Cursor_Count) {
+                return;
+            }
+            display_cursor &cursor = cursors[cursor_target];
+            size_t index = 0;
+            while(characters[index] != '\0'){
+                set_character(cursor.cursor_x, cursor.cursor_y, characters[index], 
+                    color_to_pixel(cursor.cursor_color));
+
+                // If the cursor is about to go out of bounds, return.
+                if (cursor.cursor_x + 8 < Display_Size_Width) {
+                    set_cursor_positon(cursor_target, cursor.cursor_x + 8, cursor.cursor_y);
+                } else {
+                    return;
+                }
+                index++;
+            }
+        }
+
+        /**
+         * @brief Sets the targeted cursor to the given position. Can't go out of
+         * the size of the window. It will stop before that. Any characters 
+         * written to be out of bounds will be drawn over each other.
+         * 
```

Comment:
Weird design choice, out of bounds writes should just not be displayed. What if you want to have scrolling text?

Reply:
This could be a new feature, not important for this pr.